constraint/pkg/client/client.go

```diff
@@ -253,96 +240,20 @@ func (c *Client) createBasicTemplateArtifacts(templ *templates.ConstraintTemplat
 	}, nil
 }
 
-// createTemplateArtifacts will validate the CT, create the CRD for the CT's constraints, then
-// validate and rewrite the rego sources specified in the CT.
-func (c *Client) createTemplateArtifacts(templ *templates.ConstraintTemplate) (*ctArtifacts, error) {
-	artifacts, err := c.createBasicTemplateArtifacts(templ)
-	if err != nil {
-		return nil, err
-	}
-
-	var externs []string
-	for _, field := range c.allowedDataFields {
-		externs = append(externs, fmt.Sprintf("data.%s", field))
-	}
-
-	libPrefix := templateLibPrefix(artifacts.targetHandler.GetName(), artifacts.crd.Spec.Names.Kind)
-
-	rr, err := regorewriter.New(
-		regorewriter.NewPackagePrefixer(libPrefix),
-		[]string{"data.lib"},
-		externs)
-	if err != nil {
-		return nil, fmt.Errorf("creating rego rewriter: %w", err)
-	}
-
-	entryPoint, err := parseModule(artifacts.namePrefix, artifacts.targetSpec.Rego)
-	if err != nil {
-		return nil, fmt.Errorf("%w: %v", ErrInvalidConstraintTemplate, err)
-	}
-
-	if entryPoint == nil {
-		return nil, fmt.Errorf("%w: failed to parse module for unknown reason",
-			ErrInvalidConstraintTemplate)
-	}
-
-	if err = rewriteModulePackage(artifacts.namePrefix, entryPoint); err != nil {
-		return nil, err
-	}
-
-	req := map[string]struct{}{"violation": {}}
-
-	if err = requireModuleRules(entryPoint, req); err != nil {
-		return nil, fmt.Errorf("%w: invalid rego: %v",
-			ErrInvalidConstraintTemplate, err)
-	}
-
-	rr.AddEntryPointModule(artifacts.namePrefix, entryPoint)
-	for idx, libSrc := range artifacts.targetSpec.Libs {
-		libPath := fmt.Sprintf(`%s["lib_%d"]`, libPrefix, idx)
-		if err = rr.AddLib(libPath, libSrc); err != nil {
-			return nil, fmt.Errorf("%w: %v",
-				ErrInvalidConstraintTemplate, err)
-		}
-	}
-
-	sources, err := rr.Rewrite()
-	if err != nil {
-		return nil, fmt.Errorf("%w: %v",
-			ErrInvalidConstraintTemplate, err)
-	}
-
-	var mods []string
-	err = sources.ForEachModule(func(m *regorewriter.Module) error {
-		content, err2 := m.Content()
-		if err2 != nil {
-			return err2
-		}
-		mods = append(mods, string(content))
-		return nil
-	})
-	if err != nil {
-		return nil, fmt.Errorf("%w: %v",
-			ErrInvalidConstraintTemplate, err)
-	}
-
-	return &ctArtifacts{
-		basicCTArtifacts: *artifacts,
-		modules:          mods,
-	}, nil
-}
-
 // CreateCRD creates a CRD from template.
 func (c *Client) CreateCRD(templ *templates.ConstraintTemplate) (*apiextensions.CustomResourceDefinition, error) {
 	if templ == nil {
 		return nil, fmt.Errorf("%w: got nil ConstraintTemplate",
 			ErrInvalidConstraintTemplate)
 	}
 
-	artifacts, err := c.createTemplateArtifacts(templ)
+	artifacts, err := c.createBasicTemplateArtifacts(templ)
 	if err != nil {
 		return nil, err
 	}
+	if _, _, err = local.MapModules(templ, c.allowedDataFields); err != nil {
```

Comment:
MapModules doesn't need to be called here. "MapModules" is already called from within `local.Driver.AddTemplate`, so we know we'll properly fail there. We aren't using any data from this call, so it can be safely removed.

Reply:
Removed rego validation from CreateCRD, renamed `MapModules` to `ValidateConstraintTemplate`